source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,316 @@
+    {
+          /* Loop over the entire array and see if we find a point
+           * that's within delta of what we're looking for. The data
+           * is arranged in a way that keeps the latitude constant
+           * while running over all longitudes. When we're finished
+           * with that change the latitude by one step. So if the
+           * latitude is wrong, we can definitely skip ahead a whole
+           * set of longitudes. The number of values to skip is calculated.
+           */
+
+        for (unsigned int i = 0; i <= latitudes_iso.size();)
+	    if (std::fabs(latitude - latitudes_iso[i]) <= delta)
+              {
+                if (std::fabs(longitude - longitudes_iso[i]) <= delta)
+                  return -depths_iso[i]*1000;
+                else
+                  ++i;
+              }
+         else
+              i += number_coords_depth;
+
+          Assert(false, ExcInternalError());
+          return 0;
+     }
+     
+    /**
+     * Calculate distance from the center of the Earth to the surface of WGS84.
+     */
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::radius_wgs84(const double &theta) const
+    {
+          const double eccentricity    = 8.1819190842622e-2;
+          const double semi_major_axis = 6378137.0;
+          return semi_major_axis/std::sqrt(1- eccentricity * eccentricity * std::sin(numbers::PI*theta/180)*std::sin(numbers::PI*theta/180));
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::initial_temperature (const Point<dim> &) const
+    {
+          Assert (false, ExcNotImplemented());
+          return 0;
+    }
+
+    /** Get the depth of the adiabatic boundary isotherm for the current lat/long.
+     * If above the isotherm use a linear interpolation to the surface.
+     * If below the isotherm use an increase of .5 degrees per kilometer.
+     */
+
+    template <>
+    double
+    AdiabaticBoundary<3>::initial_temperature (const Point<3> &position) const
+    {
+          const std::pair<double, double> lat_long = lat_long_from_xyz_wgs84(position);
+          const double depth                       = position.norm() - radius_wgs84(lat_long.first);
+          const double isotherm_temp               = 1673.15;
+          const double surface_temp                = 273.15;
+          const double isotherm_depth              = get_isotherm_depth(lat_long.first, lat_long.second);
+          if (depth < isotherm_depth)
+          return isotherm_temp - (depth - isotherm_depth) * 0.0005;
+          else
+          return surface_temp + ( depth/isotherm_depth)*(isotherm_temp - surface_temp);
+     } 
+
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::declare_parameters(ParameterHandler &prm)
+    {
+         prm.enter_subsection("Initial conditions");
+         {
+           prm.enter_subsection("Adiabatic boundary isotherm");
+           {
+             prm.declare_entry("Adiabatic Boundary Filename",
```

Comment:
We typically capitalize only the first word of an option (just like for section names).